backend/fastapi_structure.py

```
@@ -0,0 +1,378 @@
+        # Generate embeddings in the background
+        await embedding_service.generate_embeddings(document_id)
+        
+        return UploadResponse(
+            document_id=document_id,
+            filename=file.filename,
+            status="processing",
+            message="Document uploaded successfully and processing has begun"
+        )
+        
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)